src/moonlink/src/storage/mooncake_table/snapshot_read_output.rs

```diff
@@ -74,75 +84,57 @@ impl ReadOutput {
             .unwrap();
     }
 
-    fn resolve_temporary_files(
-        temp_files_with_index: Vec<(usize, String)>,
-        resolved_data_files: &mut [String],
-    ) {
-        for (index, file) in temp_files_with_index {
-            resolved_data_files[index] = file;
-        }
-    }
-
     /// File ordering should be preserved because the index determines the position
     /// in the final [`resolved_data_files`] array, which delete or other operations rely on.
     async fn resolve_remote_files(
         &mut self,
         object_storage_cache: Arc<dyn CacheTrait>,
         filesystem_accessor: Arc<dyn BaseFileSystemAccess>,
-        remote_files_with_index: Vec<(usize, TableUniqueFileId, String)>,
+        remote_files_entries: Vec<RemoteFileEntry>,
         resolved_data_files: &mut [String],
         cache_handles: &mut Vec<NonEvictableHandle>,
     ) -> Result<()> {
-        let chunks = remote_files_with_index.chunks(MAX_PARALLEL_OPERATIONS);
-        let mut errs = Vec::new();
-        for chunk in chunks {
-            let futures = chunk
-                .iter()
-                .map(|(index, file_id, remote_filepath)| {
-                    let cache = object_storage_cache.clone();
-                    let fs_accessor = filesystem_accessor.clone();
-                    async move {
-                        let result = cache
-                            .get_cache_entry(*file_id, remote_filepath, fs_accessor.as_ref())
-                            .await;
-                        (index, remote_filepath, result)
+        let mut results = stream::iter(remote_files_entries.into_iter())
+            .map(|remote_file_entry| {
+                let cache = object_storage_cache.clone();
+                let fs_accessor = filesystem_accessor.clone();
+                async move {
+                    let result = cache
+                        .get_cache_entry(
+                            remote_file_entry.file_id,
+                            &remote_file_entry.remote_filepath,
+                            fs_accessor.as_ref(),
+                        )
+                        .await;
+                    (
+                        remote_file_entry.index,
+                        remote_file_entry.remote_filepath,
+                        result,
+                    )
+                }
+            })
+            .buffered(MAX_PARALLEL_OPERATIONS);
```

Comment:
We can do buffer_unordered?

Reply:
Let's use buffer_unordered to improve it furtor, thanks